modules/ROOT/pages/_partials/task-flex-add-tls-context.adoc

```diff
@@ -0,0 +1,75 @@
+//tag::intro1[]
+After you xref:asm-secret-group-creation-task.adoc[create a secret group] and xref:asm-secret-group-creation-task.adoc#adding-a-keystore[add a keystore], you can add a Transport Layer Security (TLS) context and apply the TLS context to API Instances to encrypt their inbound and outbound traffic. 
+
+//end::intro1[]
+//tag::intro2[]
+Flex Gateway supports inbound and outbound TLS and mutual authentication TLS (mTLS) for both connection directions.
+
+When you add a TLS context, you can also select the ciphers to use with the TLS context and configure different inbound and outbound validation settings.
+//end::intro2[]
+
+
+//tag::intro3[]
+You can use the TLS context you store in secrets manager when you configure an HTTPS API in Flex Gateway Connected Mode.
+
+//end::intro3[]
+
+//tag::byb[]
+
+== Before You Begin
+
+Before adding the TLS context to Flex Gateway, you must complete the following tasks:
+
+* xref:anypoint-security::asm-secret-group-creation-task.adoc[Create a Secret Group]
+* xref:anypoint-security::asm-secret-group-creation-task.adoc#adding-a-keystore[Add a Keystore]
+
+//end::byb[]
+
+//tag::addContextTitle[]
+== Add a TLS Context
+
+//end::addContextTitle[]
+
+//tag::addContext[]
+Adding a TLS context to your secret group requires supplying a name, target, version, and keystore. Optionally, you can add a truststore, context experation date, ALPN Protocols, and configure the inbound and outbound settings. 
+
+[WARNING]
+====
+If you edit a secret group, including a TLS context, that is currently applied to an API Instance, you must redploy the API Instance to apply the changes. You must individual redeploy all instances affected by the change.
+
+To redeploy an API Instance:
+
+. Go to *Anypoint Platform > API Manager*.
+. Click the name of the API Instance to redploy.
+. Click *Runtime & Endpoint Configuration > Save & Apply*.
+====
```

Comment:
Is there any way to re-structure this? Including steps for another task in the warning block doesn't look good. What about something  like this: 

== Add a TLS Context
...

=== Before you Begin (or) Considerations
If you edit a secret group, including a TLS context, that is currently applied to an API Instance, you must <<redeploy-api-instance, redeploy the API Instance>> to apply the changes. You must individual redeploy all instances affected by the change.

=== Adding a TLS Context: 
To add a TLS Context: ...

[[redeploy-api-instance]]
=== Redeploying an API Instance
To redeploy an API Instance: ...

Just a suggestion